app/encoding/encoding.go

```diff
@@ -0,0 +1,39 @@
+package encoding
+
+import (
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/cosmos-sdk/std"
+	"github.com/cosmos/cosmos-sdk/x/auth/tx"
+)
+
+type InterfaceRegister func(codectypes.InterfaceRegistry)
+
+// EncodingConfig specifies the concrete encoding types to use for a given app.
+// This is provided for compatibility between protobuf and amino implementations.
+type EncodingConfig struct {
+	InterfaceRegistry codectypes.InterfaceRegistry
+	Codec             codec.Codec
+	TxConfig          client.TxConfig
+	Amino             *codec.LegacyAmino
+}
+
+// MakeEncodingConfig creates an encoding config for the app.
+func MakeEncodingConfig(regs ...InterfaceRegister) EncodingConfig {
```

Comment:
Here we are refactoring the encoding config into its own package, and instead of relying on a the [`ModuleBasics`](https://github.com/celestiaorg/celestia-app/blob/3c8376459a9c59a3a96490fc990bbafa9346cd5f/app/app.go#L125)  defined in the app, we pass a function that registers interfaces. By passing that same [method](https://github.com/celestiaorg/celestia-app/blob/3c8376459a9c59a3a96490fc990bbafa9346cd5f/x/payment/types/codec.go#L21-L41) in a module, we can use a single `MakeEncodingConfig` function everywhere in this repo

this also allows us to get rid of all the `cosmoscmd` starport code